### ISSUE TITLE
Trade License - Overall Trade License Review Needed

### ISSUE THREAD

**Issue by ABonomo21** [dhenry-KCI/FredCo-Post-Go-Live-#384]
Source:
Repository: kcigeospatial / Fred_Co_Land-Management
Ticket #: 2352
Requested By: Ashlye Bonomo
Jump to this repository

Request:
Would like to add a "resbumit" review option to the trade licence renewal review. as applicants may be missing information at renewal. Currently approved is the only option. Dan indicated that this was not the intended use for this review. Agree to re-evaluate after go-live.

image

image

**Comment by ABonomo21** [comment]
![image](https://user-images.githubusercontent.com/47611580/74251168-b8203280-4cb9-11ea-83c8-0d1c21b4a4b9.png)

![image](https://user-images.githubusercontent.com/47611580/74251190-c0786d80-4cb9-11ea-861f-fdbd3c46693a.png)

**Comment by ABonomo21** [comment]
Source:
Repository: kcigeospatial / Fred_Co_Land-Management
Ticket #: 2347
Requested By: Ashlye Bonomo
Jump to this repository

Request:
We need to ensure the grace expiration dates on all licenses have migrated properly to IPS. In some cases the grace exp was populating the same as the expire date, while another issue was identified where the license went to awaiting renewal details even though it was past the grace exp date.

**Comment by ABonomo21** [comment]
We need to move this one to current Github - Trade Licenses need work. I would like to have some time set up for Shannon, Abbie and I to meet with Dan when he is on site to walk through issues we have had on license renewals and new licenses.

**Comment by JuanitaMiles** [comment]
Will request Dustin to setup a meeting for us to kick off work on this ticket.

**Comment by JuanitaMiles** [comment]
Meeting scheduled 2/26 to discuss.

**Comment by JuanitaMiles** [comment]
Requirements as follows:

Will need to add a "resubmit" option to the renewal review. If the review is resulted with resubmit then a notification will be sent requesting more information and the application will progress to a "Awaiting Renewal Resubmit" milestone for the applicant to complete the applicant actions detail page. Once the resubmittal is completed by the applicant, the application will move back to Renewal Review for another cycle of review. Once the Renewal Review is approved, the application will progress to Active/Active and generate the new dates.

**Comment by megisys** [comment]
PLease Test this in Test and let me know your findings

**Comment by ABonomo21** [comment]
@JuanitaMiles 

can you take a look at this one- 

i went in and changed the renewal date - got the notification that the renewal period was open, did my confirmation details however, no renewal review generated for me to test if the resubmit worked... did I do something wrong?

**Comment by JuanitaMiles** [comment]
If you tested this with an electrical license you did not do anything wrong and you are correct in that it does not work properly.

Electrical License EM257849 did not progress through the Renewal Review milestone at all and therefore did not generate the Renewal review in order to test the resubmit option.

![image](https://user-images.githubusercontent.com/31770504/76458275-b036d980-63b0-11ea-9d57-fd4965b4430b.png)

![image](https://user-images.githubusercontent.com/31770504/76458329-c93f8a80-63b0-11ea-867f-0176bd76f93e.png)

**Comment by JuanitaMiles** [comment]
Plumbing Master works with the exception of the Awaiting Payment notification after the application progresses through a cycle of Awaiting Renewal Resubmit then to Active. There were no fees to be paid but a notification was sent upon approving the Renewal review.

![image](https://user-images.githubusercontent.com/31770504/76458677-466aff80-63b1-11ea-84de-5817bdeaa10f.png)

![image](https://user-images.githubusercontent.com/31770504/76458716-584ca280-63b1-11ea-81ee-fd1fe2de5248.png)

**Comment by JuanitaMiles** [comment]
Each trade license application type will need to be tested in order to test accept.

**Comment by JuanitaMiles** [comment]
Appliance Master - PA257851 - Several Problems

1) Duplicate Renewal Notifications were sent.
    
![image](https://user-images.githubusercontent.com/31770504/76459567-d2c9f200-63b2-11ea-8edb-0f2a46505322.png)

2) Resubmit option on the Renewal review was there but when resulting with resubmit the application did not progress through Awaiting Renewal Resubmit milestone and then generate a 2nd renewal review. It went straight to active.
  
![image](https://user-images.githubusercontent.com/31770504/76459772-2b00f400-63b3-11ea-8dad-a9b7e9549f60.png)

3) The same errant Awaiting Fee Payment notification was sent after resulting the renewal review and progressing to active even though there were no fees to be paid.
   
![image](https://user-images.githubusercontent.com/31770504/76459909-6dc2cc00-63b3-11ea-8e98-da6d1c995b99.png)

**Comment by JuanitaMiles** [comment]
Plumbing Misc - PJ257852 - Couple of Problems

1) Resubmit option on the Renewal review was there but when resulting with resubmit the application did not progress through Awaiting Renewal Resubmit milestone and then generate a 2nd renewal review. It went straight to active.

![image](https://user-images.githubusercontent.com/31770504/76460690-d8283c00-63b4-11ea-8020-932fae75d5a5.png)

2) The same errant Awaiting Fee Payment notification was sent after resulting the renewal review and progressing to active even though there were no fees to be paid.

![image](https://user-images.githubusercontent.com/31770504/76460845-2b01f380-63b5-11ea-9af3-79ead2e1814a.png)

**Comment by JuanitaMiles** [comment]
Utility - PU257853 - Several problems

1) No ready for renewal notification was sent.
![image](https://user-images.githubusercontent.com/31770504/76461429-415c7f00-63b6-11ea-9fe3-17b1e565a4b9.png)

2) Resubmit option on the Renewal review was there but when resulting with resubmit the application did not progress through Awaiting Renewal Resubmit milestone and then generate a 2nd renewal review. It went straight to active.

![image](https://user-images.githubusercontent.com/31770504/76461558-81236680-63b6-11ea-943a-7d145157075b.png)

3) The same errant Awaiting Fee Payment notification was sent after resulting the renewal review and progressing to active even though there were no fees to be paid.

![image](https://user-images.githubusercontent.com/31770504/76461630-a617d980-63b6-11ea-841e-17c9e791564a.png)

**Comment by megisys** [comment]
These issues have been addressed.

Thanks

Dan

**Comment by JuanitaMiles** [comment]
**Electrical - EM257854**
1) The errant Awaiting Fee Payment notification was sent after resulting the renewal review with approved and progressing to active even though there were no fees to be paid.
![image](https://user-images.githubusercontent.com/31770504/77095765-b9452d80-69e4-11ea-9ff0-c932c4848b98.png)

**Plumbing Master - PM257855**
1) The errant Awaiting Fee Payment notification was still sent after resulting the renewal review with approved and progressing to active even though there were no fees to be paid.
![image](https://user-images.githubusercontent.com/31770504/77096588-eba35a80-69e5-11ea-8353-30d791ae23cd.png)

**Appliance Master - PA257856**
1) The errant Awaiting Fee Payment notification was still sent after resulting the renewal review with approved and progressing to active even though there were no fees to be paid.
![image](https://user-images.githubusercontent.com/31770504/77097610-6d47b800-69e7-11ea-8647-63b2e99b6bba.png)

2) The Completeness Check review was resulted with Resubmit but the application is not bypassing the other status check requiring the appliance detail page to be completed. It should bypass the other status checks and progress to Awaiting More Information.
![image](https://user-images.githubusercontent.com/31770504/77106679-e1895800-69f5-11ea-9848-4b92edbe2cb7.png)
![image](https://user-images.githubusercontent.com/31770504/77106697-e8b06600-69f5-11ea-8be1-8ae29ee2d16d.png)

**Plumbing Misc - PJ257857**
1) Resubmit option on the Renewal review was there but when resulting with resubmit the application did not progress through Awaiting Renewal Resubmit milestone and then generate a 2nd renewal review. It went straight to active. Last Renewal Date did not populate when it moved to active but this could be due to the incorrect configuration of the Renewal Review resubmit.
![image](https://user-images.githubusercontent.com/31770504/77098531-e562ad80-69e8-11ea-888b-caa192e642c6.png)
![image](https://user-images.githubusercontent.com/31770504/77109290-0bdd1480-69fa-11ea-8e6e-869324edb7a6.png)

2) The errant Awaiting Fee Payment notification was still sent after resulting the renewal review with approved and progressing to active even though there were no fees to be paid.
![image](https://user-images.githubusercontent.com/31770504/77098611-04f9d600-69e9-11ea-9d39-ac0eeb6ece51.png)

**Utility - PU257858**
1) Resubmit option on the Renewal review was there but when resulting with resubmit the application did not progress through Awaiting Renewal Resubmit milestone and then generate a 2nd renewal review. It went straight to active. Approved Renewal Review for PU257863 on the first cycle and the Last Renewal Date did not populate when it moved to active but this could be due to the incorrect configuration of the Renewal Review resubmit.
![image](https://user-images.githubusercontent.com/31770504/77099487-68d0ce80-69ea-11ea-829c-06ebce03ba74.png)
![image](https://user-images.githubusercontent.com/31770504/77109577-85750280-69fa-11ea-8c51-b17712d6f2e8.png)
![image](https://user-images.githubusercontent.com/31770504/77109599-90c82e00-69fa-11ea-8081-8d96f6c9567f.png)


2) The errant Awaiting Fee Payment notification was still sent after resulting the renewal review with approved and progressing to active even though there were no fees to be paid.
![image](https://user-images.githubusercontent.com/31770504/77099576-8d2cab00-69ea-11ea-9bb0-6df4127103e4.png)

3) I have also noticed that the Utility license is the only license application type that automatically populates the Confirm License Details page at the Awaiting Renewal Details milestone with the information from the main license details page. 

Overall Issue
1) Upon approving the Renewal Review, the updated Address 2 field on the Confirm License Details page does not carry the updated value to the main Trade License details page. The other fields carry over properly. There is also inconsistency in the Address 2 field being required between the Confirm Details and Trade License details pages.
![image](https://user-images.githubusercontent.com/31770504/77107593-50b37c00-69f7-11ea-8014-bcc1549da209.png)

![image](https://user-images.githubusercontent.com/31770504/77107552-3ed1d900-69f7-11ea-9ace-9ff7b9e843f8.png)

**Comment by megisys** [comment]
Hi Juanita,  I believe i caught all the required changes. There is alot in here. Hard to ensure i got everything.. Did my best here

**Comment by JuanitaMiles** [comment]
**All Trade License Application Types:**
Electrical (EM257864); Plumbing Master - PM257865; Appliance Master - PA257866; Plumbing Misc - PJ257867; Utility - PU257868
1) The errant Awaiting Fee Payment notification was still sent after resulting the renewal review with approved and progressing to active even though there were no fees to be paid.
![image](https://user-images.githubusercontent.com/31770504/77179610-13a2c480-6a9f-11ea-93e5-73676bab52d4.png)

**Utility - PU257868**
1) The Confirm License Details page is still auto-populating with the information on the Trade License Details page upon entering the Awaiting Renewal Details milestone. This is inconsistent with every other application type. For consistency, all AP Types should be configured the same.
![image](https://user-images.githubusercontent.com/31770504/77183007-0f2cda80-6aa4-11ea-93e7-48c155064690.png)
![image](https://user-images.githubusercontent.com/31770504/77182973-03d9af00-6aa4-11ea-9826-99f6aff8a7cd.png)

**Comment by JuanitaMiles** [comment]
Tested all Trade ap types. All issues I identified now appear to be resolved. Test Accepted.
